src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -184,7 +184,103 @@ public int getTotalMessageCount() {
 	  PreparedQuery results = datastore.prepare(query);
 	  return results.countEntities(FetchOptions.Builder.withLimit(1000));
   }
-  
+
+  /**
+   * Stores Charity in datastore
+   */
+
+  public void storeCharity(Charity charity) {
+    Entity charEntity = new Entity("Charity", charity.getName());
+    charEntity.setProperty("name", charity.getName());
```

Comment:
Everything is perfect, but I suggest one small change: currently the search is case-sensitive, so it might not find type "health". what if you set the stored and searched value to all lower, or all upper? (charEntity.setProperty("name", charity.getName().toLowerCase());

Reply:
I'll do that for next week, thanks!